### ISSUE TITLE
plugin upgrade fails

### ISSUE THREAD

**Issue by hex2a** [elastic/puppet-elasticsearch#847]
* Module version: 5.2.0
* Puppet version: 4.10.0

## Bug description

instead of removing the old plugin (com.floragunn:search-guard-5:**5.4.0-12**), puppet tries to remove the new plugin (which is not even installed yet): (com.floragunn:search-guard-5:**5.4.3-12**),

the plugin that should be removed/upgraded is "search-guard-5".

```
Debug: Elasticsearch_plugin[com.floragunn:search-guard-5:5.4.3-12](provider=elasticsearch_plugin): Got com.floragunn:search-guard-5:5.4.0-12/false Expected com.floragunn:search-guard-5:5.4.3-12/false. Removing for reinstall
Debug: Executing: '/usr/share/elasticsearch/bin/elasticsearch-plugin remove com.floragunn:search-guard-5:5.4.3-12'
Error: /Stage[main]/Searchguard/Elasticsearch::Plugin[com.floragunn:search-guard-5:5.4.3-12]/Elasticsearch_plugin[com.floragunn:search-guard-5:5.4.3-12]: Could not evaluate: Execution of '/usr/share/elasticsearch/bin/elasticsearch-plugin remove com.floragunn:search-guard-5:5.4.3-12' returned 78: -> removing [com.floragunn:search-guard-5:5.4.3-12]...
ERROR: plugin [com.floragunn:search-guard-5:5.4.3-12] not found; run 'elasticsearch-plugin list' to get list of installed plugins
```

**Comment by tylerjl** [comment]
@hex2a can you provide two example manifests (one before the upgrade, one that performs the upgrade) so I can try and reproduce the issue?

**Comment by hex2a** [comment]
```
$_elasticsearch_major_version = split($facts['es_version'], '\.')[0]
$searchguard_version = '14'
$searchguard_release = "com.floragunn:search-guard-${_elasticsearch_major_version}:${facts['es_version']}-${searchguard_version}"
$sg_plugin_name = "search-guard-${_elasticsearch_major_version}"

elasticsearch::plugin { $searchguard_release :
  instances => $instance_name,
  module_dir => $sg_plugin_name,
}
```

the `es_version` fact contains the version of the currently installed `elasticsearch` package. (`5.4.0` before the upgrade, `5.4.3` or `5.5.0` after the upgrade)

**Comment by tylerjl** [comment]
@hex2a this looks like the plugin provider is confused because the format of the plugin (`vendor:name:version`) differs from the way plugins are normally indicated (`vendor/name/version`). However it _is_ a supported format to install maven plugins, so I'll put together a solution and push out a bugfix release for this.

Thank you for your report!

**Comment by tylerjl** [comment]
Should be fixed in #857, will go out in the next release.
